src/main/haskell/kore/src/Kore/Builtin/List.hs

```diff
@@ -418,27 +453,70 @@ unify
         :: (level ~ Object)
-        => SymbolOrAlias level
+        => Sort level
+        -> Seq p
+        -> SymbolOrAlias level
+        -> Seq p
+        -> p
+        -> m (expanded, proof)
+    unifyFramedRight resultSort list1 symConcat prefix2 frame2
+      | Seq.length prefix2 > Seq.length list1 =
+        return (ExpandedPattern.bottom, SimplificationProof)
+      | otherwise =
+        do
+            (prefixUnified, _) <- unifyConcrete resultSort prefix1 prefix2
+            (suffixUnified, _) <- simplifyChild frame2 listSuffix1
+            let result =
+                    (<$>)
+                        (App_ symConcat)
+                        (propagatePredicates [prefixUnified, suffixUnified])
+            return (result, SimplificationProof)
+      where
+        asBuiltinDomainList = DV_ resultSort . BuiltinDomainList
+        (prefix1, suffix1) = Seq.splitAt prefixLength list1
+          where
+            prefixLength = Seq.length prefix2
+        listSuffix1 = asBuiltinDomainList suffix1
+
+    unifyFramedLeft
+        :: level ~ Object
+        => Sort level
         -> Seq p
+        -> SymbolOrAlias level
         -> p
-        -> Sort level
         -> Seq p
         -> m (expanded, proof)
-    unifyFramed symConcat list1 v dvSort list2
-      | Seq.length list1 > Seq.length list2 =
+    unifyFramedLeft resultSort list1 symConcat frame2 suffix2
+      | Seq.length suffix2 > Seq.length list1 =
         return (ExpandedPattern.bottom, SimplificationProof)
       | otherwise =
         do
-            (prefixUnified, _) <- unifyConcrete dvSort list1 prefix2
-            (suffixUnified, _) <- simplifyChild v listSuffix2
+            (prefixUnified, _) <- simplifyChild frame2 listPrefix1
+            (suffixUnified, _) <- unifyConcrete resultSort suffix1 suffix2
             let result =
                     (<$>)
                         (App_ symConcat)
                         (propagatePredicates [prefixUnified, suffixUnified])
             return (result, SimplificationProof)
       where
-        asBuiltinDomainList = DV_ dvSort . BuiltinDomainList
-        (prefix2, suffix2) = Seq.splitAt (Seq.length list1) list2
-        listSuffix2 = asBuiltinDomainList suffix2
+        asBuiltinDomainList = DV_ resultSort . BuiltinDomainList
+        (prefix1, suffix1) = Seq.splitAt prefixLength list1
+          where
+            prefixLength = Seq.length list1 - Seq.length suffix2
+        listPrefix1 = asBuiltinDomainList prefix1
+
+    -- | Return an unsolved predicate
+    unsolved
+        :: level ~ Object
+        => PureMLPattern level variable
+        -> PureMLPattern level variable
+        -> m (expanded, proof)
+    unsolved a b =
+        let
+            unified = give symbolOrAliasSorts mkAnd a b
+            predicate = give symbolOrAliasSorts makeCeilPredicate unified
+            expanded = (give symbolOrAliasSorts pure unified) { predicate }
+        in
+            return (expanded, SimplificationProof)
```

Comment:
Maybe define
```
unsolved p = pure p { predicate = makeCeilPredicate p }
```
and call ``giveSymbolOrAliasSorts $ unsolved (a `mkAnd` b)``

Reply:
Then `unsolved` would just be a `\ceil` predicate, and not an "unsolved" unification problem.